### ISSUE TITLE
Nullability warnings are not reported inside collection expressions

### ISSUE THREAD

**Issue by thinker227** [dotnet/roslyn#69979]
**Version Used**: 4.8.0-1.23374.10 (520e2f10)

**Steps to Reproduce**:

1. Assign a collection expression containing a null literal (or some other value which may be null) to a collection type which element type is not nullable.
```cs
IEnumerable<string> xs = ["a", null, "b"];
```

[Sharplab link](https://sharplab.io/#v2:EYLgtghglgdgNAFxANwKYCcoDMCecAmIA1AD4ACADAARkCMALANwCwAUG3QMwA8dFAfFQAeAZyoBeKgG0ARBBlwqMAK4AbVYpnAZAXUZA===)

**Expected Behavior**: CS8625 (or another applicable nullability warning) is reported on `null`.

Relates to https://github.com/dotnet/roslyn/issues/66418

**Actual Behavior**: No nullability warnings are reported.

**Comment by thinker227** [comment]
Was informed this is already being worked on.

**Comment by CyrusNajmabadi** [comment]
Reactivating while the issue still repros

**Comment by jcouv** [comment]
Closing as duplicate of https://github.com/dotnet/roslyn/issues/68786
